### DISCUSSION TITLE
In ios device while connecting to trust wallet or metamask it redirects to the homepage of respective wallet

### DISCUSSION THREAD

**Discussion by akashgarg-13** [WalletConnect/walletconnect-monorepo#2605] [Developer Support]
Here is my code, when i am trying to connect to metamask or trust wallet in chrome its redirecting to the homepage of respective wallet.

 walletConnect: async (path) => {
    try {
      let account;
      const networkInfo =
        path != "/staking" ? NETWORK_INFO : NETWORK_INFOMATION;
      const chainidBlockchain = store?.getState()?.user?.chainId;
      let chain =
        path == "/staking" ? networkInfo[0]?.chain_id : chainidBlockchain;
      console.log("providerprovider", provider);
      // if (!isIOS) {
      if (provider?.chainId != chain) {
        await provider.enable();
        const walletConnectResponse = await networkChangeHelper(provider, path);
        
        if (walletConnectResponse) {
          account = await provider.enable();
        }
        // }
        account = await provider.enable();
        return account[0];
      }
      
    } catch (error) {
      provider = new WalletConnectProvider({
        rpc: {
          56: RPCURLEXCHBSC,
          1: RPCURL,
        },
        qrcode: true,
        qrcodeModalOptions: {
          mobileLinks: ["metamask", "trust"],
          desktopLinks: ["encrypted ink"],
        },
      });
    }
  },

**Comment by akashgarg-13** [comment]
The scenario is that we want users to connect through a metamask or
trustwallet, which is why QR modalOptios is used.
I have tried removing that, but then the list of multiple wallets was
displayed, and we don't want the user to interact with all those wallets,
and transactions were not executed after removing qrModalOptions .
Kindly suggest a better solution.

**Comment by finessevanes** [comment]
@akashgarg-13 can you include the SDK and versions you're using?
